yang-lsp/io.typefox.yang/src/main/java/io/typefox/yang/formatting2/YangFormatter.xtend

```diff
@@ -702,38 +703,42 @@ class MultilineStringReplacer implements ITextReplacer {
         def build(List<ILeafNode> leafNodes) {
             for (it : leafNodes) {
                 if (isHidden) {
-                    val grammarElement = grammarElement
-                    if (ML_COMMENTRule == grammarElement) {
-                        addSpace()
-                        addMultiLineComment(text)
-                    }
-                    if (SL_COMMENTRule == grammarElement) {
-                        val text = text.replace(System.lineSeparator, "")
-                        addSpace()
-                        addSingleLineComment(text)
-                        newLine()
-                    }
-                    if (WSRule == grammarElement) {
-                        if (text.contains(System.lineSeparator)) {
-                            newLine()
-                        } else {
-                            addSpace()
-                        }
-                    }
-                    if (HIDDENRule == grammarElement) {
-                        if (text.contains("+")) {
-                            addPlus()
-                        }
+                    switch grammarElement {
+	                    case ML_COMMENTRule: {
+	                        addSpace()
+	                        addMultiLineComment(text)
+	                    }
+	                    case SL_COMMENTRule: {
+	                        val text = text.replace(System.lineSeparator, "")
+	                        addSpace()
+	                        addSingleLineComment(text)
+	                        newLine()
+	                    }
+	                    case WSRule: {
+	                        if (text.contains(System.lineSeparator))
+	                            newLine()
+	                        else
+	                            addSpace()
+	                    }
+	                    case HIDDENRule: {
+	                        if (text.contains("+"))
+	                            addPlus()
+	                    }
                     }
                 } else {
                     val originalStartColumn = NodeModelUtils.getLineAndColumn(it, offset).column
                     addStringValue(text, originalStartColumn)
                 }
             }
             lines.head.prepend(firstLineIndentation, Hidden)
+            val previousLine = new Wrapper(lines.head)
             lines.tail.forEach[
                 val lineTypePrefix = prefix
+                // https://github.com/theia-ide/yang-lsp/issues/153
+                if (previousLine.get.last == '+' && lineTypePrefix.startsWith('+'))
+                	previousLine.get.removeLast()
```

Comment:
The interesting part is here :-)

Reply:
Avoiding to add the trailing `+` in the previous line seemed too complex to me, since that would require some look-ahead logic, so I'm removing it afterwards instead.